### ISSUE TITLE
Query: Owned entity without owner logic

### ISSUE THREAD

**Issue by smitpatel** [dotnet/efcore#24807]
Found this query in our tests:
```
            var subDependent1 = context.Set<Order>()
                .Include(a => a.OrderDetails.BillingAddress.OrderDetails.Order)
                .Select(o => o.OrderDetails.BillingAddress)
                .Single();
```
Order is the root
OwnsOne OrderDetails (back ref Order)
which in turns OwnsOne BillingAddress (back ref OrderDetails)

Tests used in memory so it worked in past. This throws for relational providers in 5.0
Exception
```
Unhandled exception. System.InvalidOperationException: A tracking query is attempting to project an owned entity without a corresponding owner in its result, but owned entities cannot be tracked without their owner. Either include the owner entity in the result or make the query non-tracking using 'AsNoTracking'.
   at Microsoft.EntityFrameworkCore.Query.ShapedQueryCompilingExpressionVisitor.EntityMaterializerInjectingExpressionVisitor.Inject(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.ShapedQueryCompilingExpressionVisitor.InjectEntityMaterializers(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.RelationalShapedQueryCompilingExpressionVisitor.ShaperProcessingExpressionVisitor.VisitExtension(Expression extensionExpression)
   at System.Linq.Expressions.Expression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
```

Should we support this scenario?

**Comment by AndriySvyryd** [comment]
We should, but not a priority until we get a user report

**Comment by hanslai** [comment]
I believe I am having the same issue,  is it fixed in 6.0?
